framework/src/executioners/Transient.C

```diff
@@ -281,6 +281,9 @@ Transient::init()
     //  _dt = computeConstrainedDT();
     _dt = getDT();
   }
+
+  if (!_app.isRecovering())
```

Comment:
Another thing in my mind is that I want to make the code more defensive here by adding
```
if (_dt <=0)
  mooseError("Time step size is not properly initialized");
```
Any comments?

Reply:
Do it!